--- a/drivers/misc/Makefile
+++ b/drivers/misc/Makefile
@@ -75,3 +75,6 @@ obj-$(CONFIG_MCHP_LAN966X_PCI)	+= lan966
 obj-y				+= keba/
 obj-y				+= amd-sbi/
 obj-$(CONFIG_MISC_RP1)		+= rp1/
+
+# LG6851F MTK CCCI/DPMAIF/CCMNI vendor modem stack
+obj-$(CONFIG_MTK_ECCCI_DRIVER) += mediatek/
--- a/drivers/misc/Kconfig
+++ b/drivers/misc/Kconfig
@@ -661,4 +661,7 @@ source "drivers/misc/mchp_pci1xxxx/Kconf
 source "drivers/misc/keba/Kconfig"
 source "drivers/misc/amd-sbi/Kconfig"
 source "drivers/misc/rp1/Kconfig"
+
+# LG6851F MTK CCCI/DPMAIF/CCMNI vendor modem stack
+source "drivers/misc/mediatek/Kconfig"
 endmenu
